### PR TITLE
User favorites

### PR DESCRIPTION
**Pull request by yjwangit**
Please take a review. Thank you!

### CODE REVIEW

File: app/src/components/Login.js [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
 
 import { useAuth0 } from "@auth0/auth0-react";
+import Button from "react-bootstrap/Button";
 
 const LoginButton = () => {
   const { loginWithRedirect } = useAuth0();
 
-  return <button onClick={() => loginWithRedirect()}>Log In</button>;
+  return (
+    <Button variant="light" size="sm" onClick={() => loginWithRedirect()}>
```
Comment (behnam) [inline app/src/components/Login.js:10]: If your callback method doesn't need access to any local variable, you could just directly pass the function as the callback, like this:
```
    <Button {/* ... */} onClick={loginWithRedirect}>
```

File: app/src/components/Results.js [hunk 2]
```diff
@@ -22,16 +26,28 @@ function Results({ apiUrl }) {
...
+  const getUserAllfavorites = (id) => {
+    //to query user's saved parks
+    axios
+      .get(
+        `http://localhost:4000/api/tasks/getUserFavorites?userId=${user.sub}`,
+      )
+      .then((res) => {
+        setSavedIds(res.data.data.map((item) => item.park_id)); //update savedIds
+      });
   };
```
Comment (behnam) [inline app/src/components/Results.js]: Maybe you want to put the shared parts of`getUserAllfavorites()` function in a helper module, to share between `Results.js` and `Park.js`.

File: app/src/components/Selection.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import React, { useState } from "react";
 
 //handleAreaSearch是在props中传递过来的函数，此函数在app.js中定义,需要传入statecode参数在app.js中进行接口的请求
+
 function Selection({ handleAreaSearch }) {
-  const [selectedVal, setSelectedVal] = useState("");
+  const [selectedVal, setSelectedVal] = useState("AL");
```
Comment (behnam) [inline app/src/components/Selection.js:8]: Nice!

File: server/server.mjs [hunk 4]
```diff
@@ -48,7 +48,20 @@ tasks.post("/addFavorite", async (request, response) => {
...
+    message: "success",
+    code: 201,
+    data: data,
+  });
+});
+
+tasks.delete("/delUserFavorite", async (request, response) => {
+  const { userId, parkId } = request.query; //"delusefavorites"是delete的接口地址
+  await db.delUserFavorite(userId, parkId);
+  response.status(201).json({
```
Comment (behnam) [inline server/server.mjs:61]: For a Delete action, you probably want to return HTTP 200, instead of HTTP 201.

**Review by Pidgett — COMMENTED** [review]
This looks great so far! I'm looking forward to using your app to plan my next mini-vacation!

**Review by Sophia-SR — APPROVED** [review]
Great job Yingjie! Your code looks great, very clean and so far everything seems to be working well!

**Pull request merged**